aten/src/ATen/native/quantized/cpu/qnnpack/src/conv-run.cc

```diff
@@ -497,7 +494,10 @@ enum pytorch_qnnp_status qnnpackConv(
           .m = input_size,
           .k = conv_p.group_input_channels,
           .a_stride = input_pixel_stride,
-          .multiplier = (int32_t)-conv_p.kernel_zero_point,
+          // XZP kernels are not supporting per channel quant.
+          // We dont really use XZP kernels ATM.
+          // Thus assigning the zero point of first channel.
+          .multiplier = (int32_t)-conv_p.kernel_zero_points[0],
```

Comment:
I don't think this cast is needed here.  Since an int32_t can represent any value that a uint8_t can, I think you should be OK without the cast.

Reply:
I was just moving code around. Did not put the cast there myself. And I think it existed here by copying from the original operator-run.cc. So I am gonna leave it here.